ddtrace/mocktracer/mocktracer.go

```diff
@@ -67,7 +72,23 @@ func (t *mocktracer) StartSpan(operationName string, opts ...ddtrace.StartSpanOp
 	for _, fn := range opts {
 		fn(&cfg)
 	}
-	return newSpan(t, operationName, &cfg)
+	span := newSpan(t, operationName, &cfg)
+
+	t.Lock()
+	defer t.Unlock()
+	t.openSpans[span.SpanID()] = span
```

Comment:
@gbbr removing the on-demand initialization of `openSpans` here causes a number of tests in mocktracer_test and mockspan_test to fail because a lot of them just use `mocktracer` directly (instead of going through `mocktracer.Start`) so the map ends up being nil.

Reply:
I see. Maybe it's a good time to introduce a `newMockTracer` function then? I hope it's not too much to ask. We can also drop the `defer` and just wrap the assignment like in my example.
```go
	t.Lock()
	t.openSpans[span.SpanID()] = span
	t.Unlock()
```